### DISCUSSION TITLE
Send metrics from vector to prometheus hosted on cloud

### DISCUSSION THREAD

**Discussion by uni-pooja-laad** [vectordotdev/vector#21775] [Q&A] [answered]
Hi team,
I am trying to send metrics from vector to Prometheus which is on cloud. Below is the configuration I am using:
transforms:
aggregate_metrics:
    type: aggregate
    inputs: [datadog_agent.metrics]
    interval_ms: 10000  
sinks:
 prom_metrics:
      type: prometheus_remote_write
      inputs: [aggregate_metrics]
      endpoint: "https://prometheus-prod-<region>.grafana.net"
      auth:
        strategy: basic
        user: "<username>"
        password: <token>
The error here is : 2024-11-08T06:59:19.740100Z ERROR vector::topology::builder: msg="Healthcheck failed." error=Unexpected status: 405 Method Not Allowed component_kind="sink" component_type="prometheus_remote_write" component_id=prom_metrics
Even I tried by adding suffix like: /api/prom/push or /api/v1/write etc. but seems its either 400 or 405 error.

**Comment by jszwedko** [comment, marked as answer]
Hi @uni-pooja-laad !

I think you are hitting this bug: https://github.com/vectordotdev/vector/issues/8279. For now, I'd suggest disabling the healthcheck for that sink:

```yaml
healthcheck:
  enabled: false
```

**Reply by uni-pooja-laad** [reply]
I believe this is just to avoid error. But metrics are still not getting into Prometheus.

**Reply by uni-pooja-laad** [reply]
Also, Do I need to use any exporter in between agent metrics and prom_remote _write? Does vector provides any inbuilt prometheus exporter to get metrics from datadog agent and send it to  prometheus_remote_write sink ?

**Reply by jszwedko** [reply]
> Also, Do I need to use any exporter in between agent metrics and prom_remote _write? Does vector provides any inbuilt prometheus exporter to get metrics from datadog agent and send it to prometheus_remote_write sink ?

There is a `prometheus_exporter` source you can use to scrape metrics from a Prometheus endpoint.

Are you sure you have the right Grafana Cloud `endpoint` configured for remote write? I think it may be missing a path.

**Reply by uni-pooja-laad** [reply]
yes I am using correct URL which is provided by grafana integration page.
Above I might have cropped it.
But this is the exact URL I am using https://prometheus-prod-13-prod-us-east-0.grafana.net/api/prom.

**Reply by uni-pooja-laad** [reply]
I am getting metrics from datadog_agent and using vector as an aggregator. I want to send metrics to the prometheus endpoint which is hosted on cloud. I did same task to send logs using loki sink and it is working fine. But not sure if prometheus need any extra transform or exporter.

**Reply by jszwedko** [reply]
I think you want `/api/prom/push` based on https://grafana.com/docs/grafana-cloud/send-data/alloy/reference/components/prometheus/prometheus.remote_write/#send-metrics-to-a-managed-service

**Reply by uni-pooja-laad** [reply]
Tried that as well:
https://prometheus-prod-13-prod-us-east-0.grafana.net/api/prom/push
Error is :
2024-11-15T15:28:59.607707Z ERROR sink{component_kind="sink" component_id=prom_metrics component_type=prometheus_remote_write}:request{request_id=1}: vector::sinks::util::retries: Not retriable; dropping the re
quest. reason="Http status: 400 Bad Request" internal_log_rate_limit=true

**Reply by jszwedko** [reply]
Ah, strange. Do you see any logs in Grafana Cloud indicating why the incoming request was considered invalid? There could be some clues there. I believe the path you have is now correct.
